Framework/CurveFitting/src/Functions/SmoothTransition.cpp

```diff
@@ -0,0 +1,65 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidCurveFitting/Functions/SmoothTransition.h"
+#include "MantidAPI/FunctionFactory.h"
+
+#include <cmath>
+
+namespace Mantid::CurveFitting::Functions {
+
+using namespace CurveFitting;
+
+using namespace Kernel;
+
+using namespace API;
+
+DECLARE_FUNCTION(SmoothTransition)
+
+void SmoothTransition::init() {
+  declareParameter("A1", 0.00, "maximum curve");
+  declareParameter("A2", 0.1, "Flat Background");
+  declareParameter("Midpoint", 100.0, "Sigmoid Midpoint");
+  declareParameter("GrowthRate", 1.0, "Growth rate");
+}
+
+void SmoothTransition::function1D(double *out, const double *xValues, const size_t nData) const {
+  const double a1 = getParameter("A1");
+  const double a2 = getParameter("A2");
+  const double midpoint = getParameter("Midpoint");
+  const double gr = getParameter("GrowthRate");
+
+  for (size_t i = 0; i < nData; i++) {
+    out[i] = a2 + (a1 - a2) / (exp((xValues[i] - midpoint) / gr) + 1);
```

Comment:
I have had a look into this and the descriptions for a1 and a2 are the wrong way around. Also instead of saying flat background and maximum it would be better to say:


a1: the limit of the function as x tends to zero
a2: the limit of the function as x tends to infinity 

you can see these by placing considering the limits (I will do a talk about it next week). Basically the exponential goes to zero when x is small and becomes large as x becomes larger. Then you can say these terms are much larger/smaller than the others so lets ignore them